bentoml/deployment/sagemaker/__init__.py

```diff
@@ -0,0 +1,286 @@
+            self.bento_service.name + '-' + self.bento_service.version + '-configuration')
+
+    def deploy(self):
+        """Deploy BentoML service to AWS Sagemaker.
+        Your AWS credential must have the correct permissions for sagemaker and ECR
+
+        1. Create docker image and push to ECR
+        2. Create sagemaker model base on the ECR image
+        3. Create sagemaker endpoint configuration base on sagemaker model
+        4. Create sagemaker endpoint base on sagemaker endpoint configuration
+
+        :param archive_path: Path
+        :param additional_info: Dict
+        :return: String, location to the output snapshot's path
+        """
+        snapshot_path = generate_bentoml_deployment_snapshot_path(
+            self.bento_service.name, self.bento_service.version, 'aws-sagemaker')
+        shutil.copytree(self.archive_path, snapshot_path)
+        with open(os.path.join(snapshot_path, 'nginx.conf'), 'w') as f:
+            f.write(DEFAULT_NGINX_CONFIG)
+        with open(os.path.join(snapshot_path, 'wsgi.py'), 'w') as f:
+            f.write(DEFAULT_WSGI_PY)
+        with open(os.path.join(snapshot_path, 'serve'), 'w') as f:
+            f.write(DEFAULT_SERVE_SCRIPT)
+
+        # We want to give serve '755' permission. Since chmod take octal number, 755 => 493
+        permission = '755'
+        octal_permission = int(permission, 8)
+        os.chmod(os.path.join(snapshot_path, 'serve'), octal_permission)
+
+        execution_role_arn = get_arn_role_from_current_user()
+        ecr_image_path = create_push_image_to_ecr(self.bento_service, snapshot_path)
+        sagemaker_model_info = {
+            "ModelName": self.model_name,
+            "PrimaryContainer": {
+                "ContainerHostname": self.model_name,
+                "Image": ecr_image_path,
+                "Environment": {
+                    "API_NAME": self.api.name
+                }
+            },
+            "ExecutionRoleArn": execution_role_arn,
+        }
+        logger.info('Creating sagemaker model %s', self.model_name)
+        create_model_response = self.sagemaker_client.create_model(**sagemaker_model_info)
+        logger.info('AWS create model response: %s', create_model_response)
+
+        production_variants = [{
+            "VariantName": self.bento_service.name,
+            "ModelName": self.model_name,
+            "InitialInstanceCount": self.instance_count,
+            "InstanceType": self.instant_type,
+        }]
+        logger.info('Creating sagemaker endpoint %s configuration', self.endpoint_config_name)
+        create_endpoint_config_response = self.sagemaker_client.create_endpoint_config(
+            EndpointConfigName=self.endpoint_config_name, ProductionVariants=production_variants)
+        logger.info('AWS create endpoint config response: %s', create_endpoint_config_response)
+        logger.info('Creating sagemaker endpoint %s', self.bento_service.name)
```

Comment:
maybe add a blank line and a comment like "# Creating sagemaker endpoint" between L234 and L235?

Reply:
done